tool/src/test/java/org/wildfly/security/tool/CredentialStoreCommandTest.java

```diff
@@ -131,6 +139,210 @@ public void testAliasesList() {
+        Assert.assertEquals(RSA_ALGORITHM, keyPair.getPublic().getAlgorithm());
+    }
+
+    @Test
+    public void testGenerateKeyPairDSA() throws CredentialStoreException {
+        String storageLocation = getStoragePathForNewFile();
+        String storagePassword = "cspassword";
+        String aliasName = "testalias";
+        String algorithm = DSA_ALGORITHM;
+        String size = "2048";
+
+        String[] args = { "--location=" + storageLocation, "--create", "--generate-key-pair", aliasName, "--algorithm",
+                algorithm, "--size", size, "--summary", "--password", storagePassword };
+        executeCommandAndCheckStatus(args);
+
+        CredentialStore store = getCredentialStoreStorageFromExistsFile(storageLocation, storagePassword);
+        KeyPair keyPair = store.retrieve(aliasName, KeyPairCredential.class).getKeyPair();
+        Assert.assertNotNull(keyPair);
+        Assert.assertNotNull(keyPair.getPrivate());
+        Assert.assertNotNull(keyPair.getPublic());
+        Assert.assertEquals(DSA_ALGORITHM, keyPair.getPrivate().getAlgorithm());
+        Assert.assertEquals(DSA_ALGORITHM, keyPair.getPublic().getAlgorithm());
+    }
+
+    @Test
+    public void testGenerateKeyPairECDSA() throws CredentialStoreException {
+        String storageLocation = getStoragePathForNewFile();
+        String storagePassword = "cspassword";
+        String aliasName = "testalias";
+        String algorithm = EC_ALGORITHM;
+        String size = "571";
+
+        String[] args = { "--location=" + storageLocation, "--create", "--generate-key-pair", aliasName, "--algorithm",
+                algorithm, "--size", size, "--summary", "--password", storagePassword };
+        executeCommandAndCheckStatus(args);
+
+        CredentialStore store = getCredentialStoreStorageFromExistsFile(storageLocation, storagePassword);
+        KeyPair keyPair = store.retrieve(aliasName, KeyPairCredential.class).getKeyPair();
+        Assert.assertNotNull(keyPair);
+        Assert.assertNotNull(keyPair.getPrivate());
+        Assert.assertNotNull(keyPair.getPublic());
+        Assert.assertEquals(EC_ALGORITHM, keyPair.getPrivate().getAlgorithm());
+        Assert.assertEquals(EC_ALGORITHM, keyPair.getPublic().getAlgorithm());
+    }
+
+    @Test
+    public void testExportPublicKey() {
+        String storageLocation = getStoragePathForNewFile();
+        String storagePassword = "cspassword";
+        String aliasName = "testalias";
+
+        String[] createArgs = { "--location=" + storageLocation, "--create", "--generate-key-pair", aliasName, "--summary",
+                "--password", storagePassword };
+        executeCommandAndCheckStatus(createArgs);
+
+        String[] exportArgs  = { "--location=" + storageLocation, "--export-key-pair-public-key", aliasName, "--summary",
+                "--password", storagePassword };
+        executeCommandAndCheckStatus(exportArgs);
```

Comment:
Could `executeCommandAndCheckStatusAndGetOutput` be used here to verify the output of the command is as expected?

Reply:
I've updated it to verify the output is as expected, thanks!